File: lib/src/models/appearance.dart
import 'package:meta/meta.dart';

/// Represents a line width setting in MusicXML.
@immutable
class LineWidth {
  /// The type of line (e.g., "light barline", "heavy barline", "beam", etc.).
  final String type;

  /// The width value.
  final double width;

  /// Creates a new [LineWidth] instance.
  const LineWidth({
    required this.type,
    required this.width,
  });
}

/// Represents a note size setting in MusicXML.
@immutable
class NoteSize {
  /// The type of note (e.g., "cue", "grace", etc.).
  final String type;

  /// The size value as a percentage.
  final double size;

  /// Creates a new [NoteSize] instance.
  const NoteSize({
    required this.type,
    required this.size,
  });
}

/// Represents appearance settings in a MusicXML document.
@immutable
class Appearance {
  /// Line width settings for different elements.
  final List<LineWidth> lineWidths;

  /// Note size settings for different types of notes.
  final List<NoteSize> noteSizes;

  /// Creates a new [Appearance] instance.
  const Appearance({
    this.lineWidths = const [],
    this.noteSizes = const [],
  });
}

File: MusicXMLParser/Models/Appearance.cs
using System.Collections.Generic;
using System.Linq;

namespace MusicXMLParser.Models
{
    /// <summary>
    /// Represents a line width setting in MusicXML.
    /// </summary>
    public class LineWidth
    {
        /// <summary>
        /// The type of line (e.g., "light barline", "heavy barline", "beam", etc.).
        /// </summary>
        public string? Type { get; }

        /// <summary>
        /// The width value.
        /// </summary>
        public double Width { get; }

        /// <summary>
        /// Creates a new <see cref="LineWidth"/> instance.
        /// </summary>
        public LineWidth(string? type, double width)
        {
            Type = type;
            Width = width;
        }
    }

    /// <summary>
    /// Represents a note size setting in MusicXML.
    /// </summary>
    public class NoteSize
    {
        /// <summary>
        /// The type of note (e.g., "cue", "grace", etc.).
        /// </summary>
        public string? Type { get; }

        /// <summary>
        /// The size value as a percentage.
        /// </summary>
        public double Size { get; }

        /// <summary>
        /// Creates a new <see cref="NoteSize"/> instance.
        /// </summary>
        public NoteSize(string? type, double size)
        {
            Type = type;
            Size = size;
        }
    }

    /// <summary>
    /// Represents appearance settings in a MusicXML document.
    /// </summary>
    public class Appearance
    {
        /// <summary>
        /// Line width settings for different elements.
        /// </summary>
        public List<LineWidth> LineWidths { get; }

        /// <summary>
        /// Note size settings for different types of notes.
        /// </summary>
        public List<NoteSize> NoteSizes { get; }

        /// <summary>
        /// Creates a new <see cref="Appearance"/> instance.
        /// </summary>
        public Appearance(List<LineWidth>? lineWidths = null, List<NoteSize>? noteSizes = null)
        {
            LineWidths = lineWidths ?? new List<LineWidth>();
            NoteSizes = noteSizes ?? new List<NoteSize>();
        }
    }
}
